src/commands/game_options/goal.ts

```diff
@@ -66,88 +69,166 @@ export default class GoalCommand implements BaseCommand {
+                            messageContext.guildID,
+                            "command.goal.failure.goalExceeded.title"
+                        ),
+                        description: LocalizationManager.localizer.translate(
+                            messageContext.guildID,
+                            "command.goal.failure.goalExceeded.description"
+                        ),
+                    },
+                    interaction
+                );
                 return;
             }
 
             if (gameSession.gameType === GameType.ELIMINATION) {
                 logger.warn(
                     `${getDebugLogHeader(
-                        message
+                        messageContext
                     )} | Game option conflict between goal and ${
                         gameSession.gameType
                     } gameType.`
                 );
 
-                sendErrorMessage(MessageContext.fromMessage(message), {
-                    title: LocalizationManager.localizer.translate(
-                        message.guildID,
-                        "misc.failure.gameOptionConflict.title"
-                    ),
-                    description: LocalizationManager.localizer.translate(
-                        message.guildID,
-                        "command.goal.failure.gameOptionConflict.description",
-                        {
-                            elimination: `\`${GameType.ELIMINATION}\``,
-                            goal: "`goal`",
-                            classic: `\`${GameType.CLASSIC}\``,
-                            teams: `\`${GameType.TEAMS}\``,
-                        }
-                    ),
-                });
+                sendErrorMessage(
+                    messageContext,
+                    {
+                        title: LocalizationManager.localizer.translate(
+                            messageContext.guildID,
+                            "misc.failure.gameOptionConflict.title"
+                        ),
+                        description: LocalizationManager.localizer.translate(
+                            messageContext.guildID,
+                            "command.goal.failure.gameOptionConflict.description",
+                            {
+                                elimination: `\`${GameType.ELIMINATION}\``,
+                                goal: "`goal`",
+                                classic: `\`${GameType.CLASSIC}\``,
+                                teams: `\`${GameType.TEAMS}\``,
+                            }
+                        ),
+                    },
+                    interaction
+                );
+
                 return;
             }
         }
 
-        await guildPreference.setGoal(userGoal);
+        const reset = userGoal === null;
+
+        if (reset) {
+            await guildPreference.reset(GameOption.GOAL);
+            await sendOptionsMessage(
```

Comment:
I think this can be removed, otherwise it will send the options message twice on reset